src/tests/reader/test_reader.cairo

```diff
@@ -0,0 +1,435 @@
+use starknet::{ContractAddress, contract_address_const};
+
+use satoru::data::data_store::{IDataStoreDispatcher, IDataStoreDispatcherTrait};
+use satoru::role::role_store::{IRoleStoreDispatcher, IRoleStoreDispatcherTrait};
+use satoru::role::role;
+use satoru::order::order::{Order, OrderType, OrderTrait, DecreasePositionSwapType};
+use satoru::tests_lib::{setup, teardown};
+use satoru::utils::span32::{Span32, Array32Trait};
+use satoru::market::market::{Market};
+use satoru::reader::reader;
+use snforge_std::{PrintTrait, declare, start_prank, stop_prank, ContractClassTrait};
+use poseidon::poseidon_hash_span;
+use satoru::deposit::deposit::{Deposit};
+use satoru::withdrawal::withdrawal::{Withdrawal};
+use satoru::position::position::{Position};
+use satoru::data::keys;
+use satoru::price::price::{Price, PriceTrait};
+
+#[test]
+fn given_normal_conditions_when_get_market_then_works() {
+    //
+    // Setup
+    //
+    let (caller_address, role_store, data_store) = setup();
+
+    let key: ContractAddress = 123456789.try_into().unwrap();
+    let mut market = Market {
+        market_token: key,
+        index_token: 11111.try_into().unwrap(),
+        long_token: 22222.try_into().unwrap(),
+        short_token: 33333.try_into().unwrap(),
+    };
+    start_prank(role_store.contract_address, caller_address);
+    role_store.grant_role(caller_address, role::MARKET_KEEPER);
+    stop_prank(role_store.contract_address);
+
+    // Test logic
+
+    data_store.set_market(key, 0, market);
+
+    let market_by_key = reader::get_market(data_store, key);
+    assert(market_by_key == market, 'Invalid market by key');
+
+    teardown(data_store.contract_address);
+}
+
+#[test]
+fn given_normal_conditions_when_get_market_by_salt_then_works() {
+    //
+    // Setup
+    //
+    let (caller_address, role_store, data_store) = setup();
+
+    let key: ContractAddress = 123456789.try_into().unwrap();
+    let mut market = Market {
+        market_token: key,
+        index_token: 11111.try_into().unwrap(),
+        long_token: 22222.try_into().unwrap(),
+        short_token: 33333.try_into().unwrap(),
+    };
+    start_prank(role_store.contract_address, caller_address);
+    role_store.grant_role(caller_address, role::MARKET_KEEPER);
+    stop_prank(role_store.contract_address);
+
+    let salt = poseidon_hash_span(array!['SATORU_MARKET', 0, 0, 0, 0].span());
+
+    // Test logic
+
+    data_store.set_market(key, salt, market);
+    let market_by_key = reader::get_market(data_store, key);
```

Comment:
Shouldn't you use get_market_by_salt in this test ?

Reply:
yes, you are right  I will fix it